asdf/_tests/test_deprecated.py

```diff
@@ -146,3 +146,15 @@ def test_AsdfFile_ignore_implicit_conversion_deprecation(value):
 def test_walk_and_modify_ignore_implicit_conversion_deprecation(value):
     with pytest.warns(AsdfDeprecationWarning, match="ignore_implicit_conversion is deprecated"):
         asdf.treeutil.walk_and_modify({}, lambda obj: obj, ignore_implicit_conversion=value)
+
+
+@pytest.mark.parametrize("memmap", [True, False, asdf._asdf.NotSet])
```

Comment:
I think parametrizing "copy_arrays" would be more useful here, making sure the warning is issued for both `True` and `False` and not issued for the default (not provided).